ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Protocol.hs

```diff
@@ -408,9 +427,8 @@ mkShelleyGlobals :: TPraosParams -> SL.Globals
 mkShelleyGlobals TPraosParams {..} = SL.Globals {
       epochInfo         = tpraosEpochInfo
     , slotsPerKESPeriod = tpraosSlotsPerKESPeriod
-      -- TODO where does 3 * k come from?
-    , slotsPrior        = 3 * k
-    , startRewards      = 3 * k
+    , stabilityWindow   = ceiling $ 3 * (toRational f / fromIntegral k)
+    , randomnessStabilisationWindow = ceiling $ 4 * (toRational f / fromIntegral k)
```

Comment:
We also have this in `Ouroboros.Consensus.Shelley.Ledger.Ledger`, but slightly differently, e.g., with a link to the doc. It would be good to only do this in a single place.

Reply:
Agree, but let's do in a separate PR